src/utils.ts

```diff
@@ -53,20 +53,16 @@ export function toTypeName(str: string): string {
 
 export function mergeSchema(a: any, b: any): any {
     Object.keys(b).forEach((key: string) => {
-        if (a[key] == null) {
-            a[key] = b[key];
+        const value = b[key];
+        if (typeof value !== typeof a[key]) {
```

Comment:
Nit: to match the prior behavior, should the `if` also check for `a[key] != null`?  Otherwise it would write the debug message if (for example) `a[key]` was `undefined`, which is not really a type mismatch.